### PR TITLE
pdn: fix repeated creation of rtrees in Shape::cut

### PR DESCRIPTION
**Pull request by gadfort**
Fixes:
- Filtering method to use boost filtering instead and leaving the obs rtree alone. This will likely be possible in several other places to help speed things up.

Replaces: #2795

### CODE REVIEW

**Comment by gadfort** [comment]
@oharboe FYI, I think this will have similar performance to your solution.

File: src/pdn/src/shape.cpp [hunk 1]
```diff
@@ -627,17 +636,11 @@ const odb::Rect FollowPinShape::getMinimumRect() const
...
-      // followpins can ignore grid level obstructions
-      continue;
-    }
-    filtered_obstructions.insert({box, shape});
-  }
-
-  return Shape::cut(filtered_obstructions, replacements);
+  return Shape::cut(
+      obstructions, replacements, [](const ShapeValue& other) -> bool {
+        // followpins can ignore grid level obstructions
```
Comment (maliberty) [inline src/pdn/src/shape.cpp:641]: Would you explain this a bit further?

**Review by maliberty — APPROVED** [review]

File: src/pdn/src/shape.cpp (hunk 1 above)
Comment (gadfort) [reply src/pdn/src/shape.cpp:641]: I added a little more of a comment

**Comment by oharboe** [comment]
> @oharboe FYI, I think this will have similar performance to your solution.

Yes, if anything, it is marginally faster. I measured 9 seconds and with this change, I saw 7 seconds. Previously the same PDN insertion took 424 seconds. So 40x speedup, which is not nothing :-)

**Comment by gadfort** [comment]
@oharboe thanks for pointing out the inefficiency.

**Pull request merged**